packages/mui-material/src/Button/Button.js

```diff
@@ -295,9 +296,26 @@ const ButtonEndIcon = styled('span', {
   ...commonIconStyles(ownerState),
 }));
 
+const addClassNameBasedOnPosition = (buttonGroupButtonContext) => {
+  if (
+    buttonGroupButtonContext.firstButtonClassName &&
+    buttonGroupButtonContext.lastButtonClassName
+  ) {
+    return '';
+  }
+  if (buttonGroupButtonContext.firstButtonClassName) {
+    return buttonGroupButtonContext.firstButtonClassName;
+  }
+  if (buttonGroupButtonContext.lastButtonClassName) {
+    return buttonGroupButtonContext.lastButtonClassName;
+  }
+  return buttonGroupButtonContext.middleButtonClassName;
+};
```

Comment:
What do you think about moving this logic to `ButtonGroup` so instead of returning

```ts
interface ButtonGroupButtonContextType {
  firstButtonClassName?: string;
  lastButtonClassName?: string;
  middleButtonClassName?: string;
}
```

It just returns

```ts
interface ButtonGroupButtonContextType {
  positionClassName?: string;
}
```

And then we consume that in `Button`:

```ts
const positionClassName = buttonGroupButtonContext?.positionClassName || ''
```

I think it is better that way, as logic regarding the button group is kept in the button group files.

Reply:
Makes sense. Also, instead of keeping object in the context value, I kept only the class name string.